zephyr/zephyr.cmake

```diff
@@ -101,18 +101,16 @@ if(CONFIG_PICOLIBC_USE_MODULE)
   # Fetch zephyr compile flags from interface
   get_property(PICOLIBC_EXTRA_COMPILE_OPTIONS TARGET zephyr_interface PROPERTY INTERFACE_COMPILE_OPTIONS)
 
-  # Enable POSIX APIs
-
-  # Link to C library and libgcc (on non-armclang toolchains)
-
-  if(${ZEPHYR_TOOLCHAIN_VARIANT} STREQUAL "armclang")
-    zephyr_link_libraries(c)
-  else()
-    zephyr_link_libraries(c -lgcc)
-  endif()
+  # Tell Zephyr about the module built picolibc library to link against.
```

Comment:
I managed to create a hack using a file named '-lc' -- cmake then happily added '-lc' to the link command line, and the linker resolved that using the search path specified in picolibc.specs. Otherwise, it seems like this code would need to replicate the multilib search path?

Reply:
Not sure I follow, so hopefully you can clarify a bit on the purpose of the hack :slightly_smiling_face: 

In picolibc Zephyr module then we are building picolibc from source,most likely because it's not part of the toolchain.
If picolibc is not already part of the toolchain, then there's probably not a `picolibc.specs` neither, for example LLVM.
So what is the reason or value in being able to link using `-lc` vs absolute path to the C library we have just compiled ?

Also, when linking to a picolibc as a library we have just compiled, with the flags for the given architecture inside the build folder, then why is multilib search path a benefit in this case ?
We are building for just a single arch in the current build folder where also the picolibc library will be created.
Building for LLVM or Arm Compiler 6 (armclang) would, afaict, not benefit from specs file usage and multilib search. 

The existing linking with picolibc as pre-built library provided by the toolchain is still handled here:
Default use of `-lc`:
https://github.com/zephyrproject-rtos/zephyr/blob/51a910e326b97f7382d464d5cd343665bff7abdd/cmake/linker/ld/linker_libraries.cmake#L7

Specifying `picolibc.specs`:
https://github.com/zephyrproject-rtos/zephyr/blob/51a910e326b97f7382d464d5cd343665bff7abdd/lib/libc/picolibc/CMakeLists.txt#L18-L19

Reply:
Sorry, I'll blame jet-lag for not reading this carefully enough. Somehow I thought this was for the toolchain picolibc.

Reply:
no worries. I know the feeling :smile: